From c5f357bd428a6a5c9433fa54503f09d6db1525fc Mon Sep 17 00:00:00 2001
From: Jurijs Soloveckis <jsoloveckis@maxlinear.com>
Date: Tue, 16 Apr 2024 15:00:12 +0300
Subject: [PATCH] WLANRTSYS-80194 CLONE [6.1.0] LGP-Non-SDL:PON: 2.4G WLAN
 interface is going down after reboot

Set correct center segment indexes in channel switch fallback for non-5GHz band
---
 src/ap/hostapd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index e52f5b60c..07493230f 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -5627,15 +5627,15 @@ void
 hostapd_switch_channel_fallback(struct hostapd_iface *iface,
 				const struct hostapd_freq_params *freq_params)
 {
-	int seg0_idx = 0, seg1_idx = 0;
+	u8 seg0_idx = 0, seg1_idx = 0;
 	enum oper_chan_width bw = CONF_OPER_CHWIDTH_USE_HT;
 
 	wpa_printf(MSG_DEBUG, "Restarting all CSA-related BSSes");
 
 	if (freq_params->center_freq1)
-		seg0_idx = 36 + (freq_params->center_freq1 - 5180) / 5;
+		ieee80211_freq_to_chan(freq_params->center_freq1, &seg0_idx);
 	if (freq_params->center_freq2)
-		seg1_idx = 36 + (freq_params->center_freq2 - 5180) / 5;
+		ieee80211_freq_to_chan(freq_params->center_freq2, &seg1_idx);
 
 	switch (freq_params->bandwidth) {
 	case 0:
-- 
2.43.0

